src/pydata_sphinx_theme/__init__.py

```diff
@@ -51,6 +53,31 @@ def update_config(app, env):
             " Set version URLs in JSON directly."
         )
 
+    # check the validity of the theme swithcer file
+    if isinstance(theme_options.get("switcher"), dict):
+        theme_switcher = theme_options.get("switcher")
+        if theme_switcher.get("json_url"):
+            json_url = theme_switcher["json_url"]
+            theme_switcher["version_match"]
```

Comment:
this seems extra?

Reply:
I removed the `if` statement, "json_url" should be set so if it doesn't exist it will raise an error